packages/web3-providers/src/alchemy/apis/EVM.ts

```diff
@@ -117,18 +114,12 @@ function createNonFungibleAsset(
                     ? SchemaType_EVM.ERC721
                     : SchemaType_EVM.ERC1155,
             address: metaDataResponse.contract?.address,
-            name:
-                contractMetadataResponse?.contractMetadata.name ||
-                metaDataResponse.metadata?.name ||
-                metaDataResponse.title,
+            name,
             symbol: contractMetadataResponse?.contractMetadata?.symbol ?? '',
         },
         collection: {
             chainId,
-            name:
-                contractMetadataResponse?.contractMetadata.name ||
-                metaDataResponse.metadata?.name ||
-                metaDataResponse.title,
+            name,
```

Comment:
When the collection(contract) name is absent, better not to choose the asset name as a fallback one.